plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,259 @@
+        }
+
+        if ($context === 'com_content.article') {
+            if (empty($data && $data->id)) {
+                return;
+            }
+            $this->disableCategoryField($form, $data);
+        }
+    }
+
+    /**
+     * Manipulate the generic list view
+     *
+     * @param   DisplayEvent  $event
+     *
+     * @return  void
+     *
+     * @since   4.0.0
+     */
+    public function onAfterDisplay(DisplayEvent $event)
+    {
+        if (!$this->getApplication()->isClient('administrator')) {
+            return;
+        }
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function()
+            {
+                var dropdown = document.getElementById('toolbar-status-group');
+                if (!dropdown){
+                    return;
+                }
+                var batchButton = document.getElementById('status-group-children-batch');
+                if (batchButton){
+                    batchButton.addEventListener('click', function()
+                    {
+                        var observer = new MutationObserver(function(mutations, observer) {
+                            var categorySelector = document.getElementById('batch-category-id');
+                            if (categorySelector) {
+                                categorySelector.disabled = true;
+                                observer.disconnect();
+                            }
+                        });
+
+                        observer.observe(document.body, { childList: true, subtree: true });
+                    });
+                }
+            });
+        ";
+
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
+    }
+
+    /**
+     * Check if the current plugin should execute workflow related activities
+     *
+     * @param   string  $context
+     *
+     * @return   boolean
+     *
+     * @since   DEPLOY_VERSION
+     */
+    protected function isSupported($context)
+    {
+        return in_array($context, ['com_workflow.transition', 'com_content.article']);
+    }
+
+    /**
+     * Disable the category field in the article form.
+     *
+     * @param   Form      $form  The form
+     * @param   \stdClass    $data  The data
+     *
+     * @return  void
+     *
+     * @since   DEPLOY_VERSION
+     */
+    protected function disableCategoryField(Form $form, $data)
+    {
+        // Get the current category ID value
+        $catid = $form->getValue('catid');
+
+        $form->setFieldAttribute('catid', 'readonly', 'true');
+        $form->setFieldAttribute('catid', 'value', $catid);
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function() {
+                var categoryField = document.getElementById('jform_catid');
+                if (categoryField) {
+                    categoryField.disabled = true;
+                }
+            });
+        ";
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
```

Comment:
```addScriptDeclaration``` via document is deprecated, please use:
- the webasset manager
- a JS file (+ joomla.asset.json)